### PR TITLE
Rename `IO#write_utf8` to `#write_string`.

### PR DESCRIPTION
**Pull request by straight-shoota**
> `#write_utf8` appears as if it should be used when writing UTF-8 characters. But in fact, it must be used for *any* string, even if it consists entirely of ASCII characters (they're technically also UTF-8 but we typically differentiate between (non-ASCII) UTF-8 and ASCII (UTF-8) characters for practical purposes).
> 
> I'd propose to rename that method to a more expressive name: `#write_string`

Ref https://github.com/crystal-lang/crystal/issues/11009#issuecomment-886178292

### CODE REVIEW

**Review by asterite — APPROVED** [review]
I guess the name if a bit more intuitive, though it's a bit strange that `write_string` doesn't receive a `String` as an argument, right? `write_utf8` meant a slice of bytes with utf8-encoded bytes. But this new name is better, I think. Though I guess people will still use `write` instead of `write_string`, there's no indication that when writing bytes you should be calling one or the other.

**Comment by vlazar** [comment]
> I guess the name if a bit more intuitive, though it's a bit strange that `write_string` doesn't receive a `String` as an argument, right? `write_utf8` meant a slice of bytes with utf8-encoded bytes.

Yes, `write_string(slice)` is confusing.

**Comment by straight-shoota** [comment]
I don't see it as a huge issue. `write_string` means "write these bytes, interpreted as a string".

An alternative would be `write_text` but I don't think that's better. 

@asterite In https://github.com/crystal-lang/crystal/issues/11009#issuecomment-886178292 I already mentioned that we could consider a more explicit name for `#write` as well but I see that as a separate issue.

File: src/io.cr [hunk 1]
```diff
@@ -463,8 +463,9 @@ abstract class IO
     nil
   end
 
-  # Writes a slice of UTF-8 encoded bytes to this `IO`, using the current encoding.
-  def write_utf8(slice : Bytes) : Nil
+  # Writes a slice of UTF-8 or ASCII encoded bytes to this `IO`, using the
+  # current encoding.
```
Comment (HertzDevil) [inline src/io.cr:467]:
```suggestion
  # Writes the contents of *slice*, interpreted as UTF-8 byte sequences or ASCII
  # characters, into this `IO`. The contents are transcoded into this `IO`'s
  # current encoding.
  #
  # ```
  # bytes = "你".bytes # => [228, 189, 160]
  #
  # io = IO::Memory.new
  # io.set_encoding("GB2312")
  # io.write_string(bytes)
  # io.to_slice # => Bytes[196, 227]
  #
  # "你".encode("GB2312") # => Bytes[196, 227]
  # ```
```

Comment (straight-shoota) [reply src/io.cr:467]: `String#bytes` returns `Array(UInt8)`. It should probably be `"你".to_slice`, right?

**Review by sdogruyol — APPROVED** [review]
Thank you @straight-shoota 🙏

**Comment by beta-ziliani** [comment]
Sorry for arriving late, how about changing it to `write_string_slice`? Longer but clearer.

**Comment by straight-shoota** [comment]
I guess it would be fine, but I'd still prefer `write_string`. The suffix `_slice` really doesn't add any information (we already know that it receives a slice from the type restriction).

**Pull request merged**